src/main/java/seedu/address/logic/commands/LinkCommand.java

```diff
@@ -21,58 +24,114 @@ public class LinkCommand extends Command {
 
     public static final String COMMAND_WORD = "link";
 
-    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Links all selected participants to the event. "
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Links all selected people to sponsors by "
+            + "exporting into a csv file. "
             + "Parameters: multiple index (must be a positive integer)\n"
             + "Example: " + COMMAND_WORD + " 1 2 3";
 
-    public static final String MESSAGE_SUCCESS = "Exported all selected participants";
+    public static final String MESSAGE_SUCCESS = "Exported all selected people";
+
+    private static Logger logger = Logger.getLogger("LinkCommandLogger");
+    private final List<Index> indexes;
 
-    public final int[] indexes;
 
     /**
-     * Creates a LinkCommand to link the specified {@code Person} to the Sponsors.
+     * Creates a LinkCommand to send information of the specified {@code Person} using a csv file.
      *
-     * @param indexes the indexes of the selected participants
+     * @param indexes the indexes of the selected people
      */
-    public LinkCommand(int[] indexes) {
+    public LinkCommand(Index[] indexes) {
         requireNonNull(indexes);
-        this.indexes = indexes;
+        this.indexes = Arrays.asList(indexes);
     }
 
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        Path filePath = Path.of("./selectedParticipants/list.csv");
-        try {
-            FileUtil.createFile(filePath);
-            String header = "Name, Phone, Email, Comment\n";
-            FileUtil.writeToFile(filePath, header);
-        } catch (IOException e) {
-            throw new CommandException(Messages.MESSAGE_IO_ERROR);
-        }
-        for (int i = 0; i < indexes.length; i++) {
-            List<Person> lastShownList = model.getFilteredPersonList();
-            int index = indexes[i];
+        List<Person> lastShownList = model.getFilteredPersonList();
+        validateIndexes(indexes, lastShownList);
+
+        Path filePath = Path.of("./selectedPeople/list.csv");
+        createFile(filePath);
+        writeToFile(filePath, model);
+
+        return new CommandResult(MESSAGE_SUCCESS);
+    }
 
-            if (index < 0 || index >= lastShownList.size()) {
+    /**
+     * Validates the indexes of the selected people.
+     *
+     * @param indexes the indexes of the selected people
+     * @param persons the list of all people
+     * @throws CommandException if the indexes are invalid
+     */
+    private void validateIndexes(List<Index> indexes, List<Person> persons) throws CommandException {
+        for (Index index : indexes) {
+            boolean isNegative = index.getZeroBased() < 0;
+            boolean isOver = index.getZeroBased() >= persons.size();
+            if (isNegative || isOver) {
```

Comment:
check index==0

Reply:
Index is zero-based, so zero is acceptable value. However, there was another problem with parser regex so I fixed that part.